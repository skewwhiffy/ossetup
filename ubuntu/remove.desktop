#!/usr/bin/env bash

sudo apt remove -y ubuntu-desktop
sudo apt purge -y nautilus nautilus-* gnome-power-manager gnome-screensaver gnome-termina* gnome-pane* gnome-applet* gnome-bluetooth gnome-desktop* gnome-sessio* gnome-user* gnome-shell-common zeitgeist-core libzeitgeist* gnome-control-center gnome-screenshot && sudo apt-get autoremove
sudo apt autoremove -y
