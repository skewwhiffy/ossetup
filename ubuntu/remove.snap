#!/usr/bin/env python
import subprocess


def get_installed_snaps():
    try:
        raw = subprocess.check_output(['snap', 'list'])
    except:
        return []
    raw_lines = raw.splitlines()
    return [x.decode('utf-8').split(' ')[0] for x in raw_lines][1::]


def uninstall_all_snaps():
    snaps = get_installed_snaps()
    for snap in snaps:
        subprocess.run(['sudo', 'snap', 'remove', snap])
    remaining_snaps = get_installed_snaps()
    if len(remaining_snaps) == 0:
        print('There are no more snaps. Hurray.')
        return
    if len(remaining_snaps) >= len(snaps):
        print('*** Did not manage to remove any snaps. I give up. ***')
        exit()
    if len(remaining_snaps) >= 0:
        print('There are still snaps remaining, will try again')
        uninstall_all_snaps()


def uninstall_snap():
    subprocess.check_call(['sudo', 'apt', 'purge', 'snapd', '-y'])
    subprocess.run(['sudo', 'apt-mark', 'hold', 'snapd'])

    folders_to_kill = ['~/snap', '/snap', '/var/snap', '/var/lib/snapd']
    for folder in folders_to_kill:
        print(f'Removing folder ${folder}')
        subprocess.run(['sudo', 'rm', '-rf', folder])
        
    subprocess.check_call(['./prevent.snap'])
    


if __name__ == '__main__':
    uninstall_all_snaps()
    uninstall_snap()

